### PR TITLE
samples+tests: add CBPRINTF_FP_SUPPORT to samples/tests/features that require FP formatting

### PR DESCRIPTION
**Pull request by pabigot**
Multiple tests include *printf formatting specifications for floating point output.  To make sure they work with minimal libc add support for formatting floating point to the corresponding `prj.conf` files.

Also the shell sensor module selects this functionality.

    git grep -E '%[-0-9# ]*(\.[0-9]*)?[fgeFGE]' 

(This produces false positives, which should have been filtered before the candidate set was identified.)

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

File: tests/boards/native_posix/rtc/prj.conf [hunk 1]
```diff
@@ -1,2 +1,3 @@
 CONFIG_NATIVE_POSIX_SLOWDOWN_TO_REAL_TIME=y
 CONFIG_ZTEST=y
+CONFIG_CBPRINTF_FP_SUPPORT=y
```
Comment (aescolar) [inline tests/boards/native_posix/rtc/prj.conf:3]: False positive (`posix_print_trace()` uses the host libC `printf`)

**Review by aescolar — DISMISSED** [review]
A false positive there

**Comment by avisconti** [comment]
@pabigot 
Sorry, I was commenting on my commit.
Here seems that following configuration files are missing, because the name is shub.conf, and not prj.conf. Is there any requirement on the name?

  samples/shields/x_nucleo_iks01a3/sensorhub/shub.conf
  samples/shields/x_nucleo_iks02a1/sensorhub/shub.conf

**Review by avisconti — CHANGES_REQUESTED** [review]
Two cases are missing:

samples/shields/x_nucleo_iks01a3/sensorhub/shub.conf
samples/shields/x_nucleo_iks02a1/sensorhub/shub.conf

**Comment by pabigot** [comment]
> @pabigot
> Sorry, I was commenting on my commit.
> Here seems that following configuration files are missing, because the name is shub.conf, and not prj.conf. Is there any requirement on the name?

Well, anybody maintaining samples is going to assume `prj.conf` is the one to use.  Overlay configurations are normally used to augment, rather than replace, that core configuration file.   I recommend changing that if possible.

I'll update this to do it in the `shub.conf` files after I see how CI goes.

**Comment by avisconti** [comment]
> I recommend changing that if possible.

I'll do that in the next days

**Comment by pabigot** [comment]
@aescolar @avisconti updated per your requests.  Previous run passed all tests.

**Comment by aescolar** [comment]
> @aescolar @avisconti updated per your requests. Previous run passed all tests.

@pabigot Thanks. I leave it as a +0 to allow the respective codeowners to take a look

**Review by jukkar — APPROVED** [review]
+1 to net changes

**Review by avisconti — APPROVED** [review]
LGTM now

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**